crates/networking/p2p/types.rs

```diff
@@ -347,11 +347,26 @@ impl NodeRecord {
         Ok(result)
     }
 
-    pub fn from_node(node: &Node, seq: u64, signer: &SecretKey) -> Result<Self, NodeError> {
+    pub fn from_node(
+        node: &Node,
+        seq: u64,
+        signer: &SecretKey,
+        fork_id: Option<ForkId>,
+    ) -> Result<Self, NodeError> {
```

Comment:
We should keep this function like before, and add another one for the case where `fork_id` is `Some`. That way changes are kept to this file, and API is clearer. We can reduce code duplication by moving common code to a helper in this module.